OpenRA.Mods.Common/Activities/FindResources.cs

```diff
@@ -103,7 +103,7 @@ public override Activity Tick(Actor self)
 				foreach (var n in notify)
 					n.MovingToResources(self, closestHarvestablePosition.Value, this);
 
-				return ActivityUtils.SequenceActivities(mobile.MoveTo(closestHarvestablePosition.Value, 1), new HarvestResource(self), this);
+				return ActivityUtils.SequenceActivities(mobile.MoveTo(closestHarvestablePosition.Value, 2), new HarvestResource(self), this);
```

Comment:
This change makes it so that the harvester doesn't need to move to the chosen resource cell, but potentially to the cell next to it.  This hides the symptoms of the bug you want to fix, but doesn't fix the actual problem (which is that it picks an invalid cell to harvest).  This actually makes the problem worse in the sense that the harvester may move to an empty cell next to the target (if it was at the edge of a resource field) and then try to harvest... nothing!